src/history/hs_conn.c

```diff
@@ -133,22 +133,29 @@ int
 __wt_hs_open(WT_SESSION_IMPL *session, const char **cfg)
 {
     WT_CONNECTION_IMPL *conn;
+    WT_DECL_RET;
+    WT_SESSION_IMPL *hs_session;
 
     conn = S2C(session);
 
     /* Read-only and in-memory configurations don't need the history store table. */
     if (F_ISSET(conn, WT_CONN_IN_MEMORY | WT_CONN_READONLY))
         return (0);
 
+    /* We need a real session to open the HS file. */
```

Comment:
Can we expand the comment here to clarify why we need a real session?

Reply:
I improved the comment, please check.